src/coreclr/src/jit/morph.cpp

```diff
@@ -17503,6 +17511,11 @@ void Compiler::fgPromoteStructs()
     lvaStructPromotionInfo structPromotionInfo;
     bool                   tooManyLocalsReported = false;
 
+    // Clear the structPromotionHelper, since it is used during inlining, at which point it
+    // may be conservative about looking up SIMD info.
+    // We don't want to preserve those conservative decisions for the actual struct promotion.
```

Comment:
Thanks for the explanation and the comment, I would probably rephrase it as: `since it is used during importation for inlining decisions.`

the place that uses it before we call struct promotion is:
https://github.com/dotnet/runtime/blob/334bd8491f17884536b8d64da4c2e36a50517ee4/src/coreclr/src/jit/importer.cpp#L18651-L18658